test/Java8andUp/src/org/openj9/test/constantPoolTags/ConstantPoolTagTests.java

```diff
@@ -0,0 +1,154 @@
+/*******************************************************************************
+ * Copyright (c) 2017, 2017 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https:www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https:www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code is also Distributed under one or more Secondary Licenses,
+ * as those terms are defined by the Eclipse Public License, v. 2.0: GNU
+ * General Public License, version 2 with the GNU Classpath Exception [1]
+ * and GNU General Public License, version 2 with the OpenJDK Assembly
+ * Exception [2].
+ *
+ * [1] https:www.gnu.org/software/classpath/license.html
+ * [2] http:openjdk.java.net/legal/assembly-exception.html
+ *******************************************************************************/
+package org.openj9.test.constantPoolTags;
+
+import java.lang.reflect.Field;
+import java.util.*;
+
+import org.testng.annotations.*;
+import org.testng.log4testng.Logger;
+import org.testng.*;
+
+import sun.misc.Unsafe;
+
+import org.objectweb.asm.ClassWriter;
+import org.objectweb.asm.ModuleVisitor;
+import org.objectweb.asm.Opcodes;
+
+/**
+ * Validate that classes with Java 9 CONSTANT_Module and CONSTANT_Package
+ * constant pool tags cannot be loaded, while modules with these tags can.
+ * 
+ * @author Theresa Mammarella
+ */
+@Test(groups = { "level.extended" })
+public class ConstantPoolTagTests {
+
+	private static Logger logger = Logger.getLogger(ConstantPoolTagTests.class);
+
+	private static final String JAVA_VERSION = System.getProperty("java.version");
+
+	private static final int classFileVersionJava8 = 52;
+	private static final int classFileVersionJava9 = 53; 
+
+	public static final int CONSTANT_Module = 19;
+	public static final int CONSTANT_Package = 20; 
+
+	private int getJavaVersion() {
+		return isJava8() ? classFileVersionJava8 : classFileVersionJava9;
+	}
+
+	private boolean isJava8() {
+		return JAVA_VERSION.startsWith("1.8.0");
+	}
+
+	private boolean isJava9() {
+		return JAVA_VERSION.startsWith("1.9.0");
```

Comment:
```
$ java -XshowSettings:properties
....
    java.version = 9-internal
....
```
This check will return false on current openj9 builds as version is temporarily `9-internal`.  Eventually, this the version will be `9` so the check here should be  `startsWith("1.9.0" || startsWith("9")`. 

Sorry, I wasn't clear earlier.